src/main/java/org/gridsuite/ds/server/service/DynamicSimulationWorkerService.java

```diff
@@ -0,0 +1,147 @@
+import com.powsybl.dynamicsimulation.groovy.GroovyDynamicModelsSupplier;
+import com.powsybl.dynamicsimulation.groovy.GroovyExtension;
+import com.powsybl.dynawo.DynawoProvider;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.network.store.client.NetworkStoreService;
+import com.powsybl.network.store.client.PreloadingStrategy;
+import org.gridsuite.ds.server.dto.DynamicSimulationStatus;
+import org.gridsuite.ds.server.repository.ResultEntity;
+import org.gridsuite.ds.server.repository.ResultRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.http.HttpStatus;
+import org.springframework.messaging.Message;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+import java.util.logging.Level;
+
+/**
+ * @author Abdelsalem Hedhili <abdelsalem.hedhili at rte-france.com>
+ */
+@Service
+public class DynamicSimulationWorkerService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DynamicSimulationWorkerService.class);
+
+    private static final String CATEGORY_BROKER_INPUT = DynamicSimulationWorkerService.class.getName()
+            + ".input-broker-messages";
+
+    private final ResultRepository resultRepository;
+
+    private final NetworkStoreService networkStoreService;
+
+    private final DynamicSimulationResultPublisherService resultPublisherService;
+
+    private FileSystem fileSystem = FileSystems.getDefault();
+
+    public DynamicSimulationWorkerService(NetworkStoreService networkStoreService,
+                                          ResultRepository resultRepository,
+                                          DynamicSimulationResultPublisherService resultPublisherService) {
+        this.networkStoreService = networkStoreService;
+        this.resultRepository = resultRepository;
+        this.resultPublisherService = resultPublisherService;
+    }
+
+    public Mono<DynamicSimulationResult> run(DynamicSimulationRunContext context) {
+        Objects.requireNonNull(context);
+
+        LOGGER.info("Run dynamic simulation on network {}, startTime {}, stopTime {}, modelContent: {}, modelName: {}",
+                context.getNetworkUuid(), context.getStartTime(), context.getStopTime(), context.getDynamicModelContent(), context.getDynamicModelFileName());
+        Path path = fileSystem.getPath(context.getDynamicModelFileName());
```

Comment:
Maybe as it is temporary we should not fix, but are you aware that:
 - your web service does not work if 2 runs are asked at the same time with a dynamic model file with the same name but different content
 - file is not deleted
 - clean way to implement this is to create de temporary file (Files.createTempo ...) so that you have a unique name.

Reply:
they can't have the same name since it's defined in the service as a random UUID. I didn't know about the createTemporaryFile. I'll leave it this way and remove all of this when upgrading version